src/syntax/platform.rkt

```diff
@@ -483,8 +472,9 @@
             [(list 'if rest ...) (loop rest assigns)]
             [(list (? (curryr assq assigns)) rest ...) (loop rest assigns)]
             [(list op rest ...)
-             (for ([impl (operator-all-impls op)])
-               (when (set-member? impls impl)
+             (for ([impl (in-set impls)])
+               (define pattern (cons op (map (lambda _ (gensym)) (operator-info op 'itype))))
+               (when (pattern-match (impl-info impl 'spec) pattern)
```

Comment:
This is pretty tricky, so just to check, this is basically saying we can recurse if the impl matches at least at the top level? But don't we need to, like, do the substitution and recurse on the remaining parts? So for example, suppose we only have an `fma` operation without a `*` operation. If I try to do `(+ (* x y) z)` I shouldn't try to recurse into the `*` operator, right?